addon/mixins/relationship-support-model.js

```diff
@@ -0,0 +1,50 @@
+import getRelationshipDescriptor from '@bagaar/ember-data-bagaaravel/utils/get-relationship-descriptor';
+import { assert } from '@ember/debug';
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  /**
+   * Methods
+   */
+
+  saveRelationship(relationshipName) {
+    assert(
+      'ember-data-bagaaravel: Cannot save a relationship of a newly created record.',
+      !this.isNew,
+    );
+
+    assert(
+      `ember-data-bagaaravel: "${relationshipName}" is not a valid relationship name.`,
+      getRelationshipDescriptor(this, relationshipName),
+    );
+
+    assert(
+      `ember-data-bagaaravel: "${relationshipName}" relationship can not be serialized.`,
+      this.canSerializeRelationship(relationshipName),
+    );
+
+    return this.save({
+      adapterOptions: {
+        relationshipName,
+      },
+    });
+  },
+
+  saveRelationships(...relationshipNames) {
+    assert(
```

Comment:
Or should I remove this one because `saveRelationship` already checks this?

Reply:
No, I think it's ok like this.